src/bsoncxx/json.cpp

```diff
@@ -274,7 +274,12 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
+    if (!value.type()) {
+        // TODO: use bsoncxx error category
+        throw std::runtime_error("invalid value");
```

Comment:
i mean specifically in the overload `std::string to_json(document::view view)`. As Tess mentioned it may be worth catching the exception and then doing something (like crashing the process) as that should be impossible, I think.